02-oop-inheritance/track.js

```diff
@@ -0,0 +1,64 @@
+Track = function () {
+
+		var init = function(title, artist, duration){
+			var title = title;
+			var artist = artist;
+			var duration = duration;
+			var observable = new Observable(); 
+		},
+
+		addObserver = function (observer){
+			this.observable.addObserver(observer);
+		}; 
+
+		return {
+			init: init,
+			addObserver: addObserver,
+
+			play : function(){
+				console.log("Playing "+this.title);
+			},
+			stop : function(){0
+				console.log(this.title+" Stopped playing");
+			},
+			set : function(attr, value){
+				var temp = attr;
+				this[temp] = value;
+			},
+			get : function(attr){
+				var temp = attr;	
```

Comment:
The init method will not work, you are setting the 3 variables as private not allowing you to reach them. You might need to refactor this.